### PR TITLE
Chat notifier plugin

### PR DESCRIPTION
**Pull request by Reasel**
# Chat notifier
This addition will add a text field similar to grounditem's highlighted items list where any of the given messages entered are being searched for. If any message appears the player will be notified and told what 
message was found.

## Option: Notified messages
These are the comma separated messages to be notified of. In the demo you can see: Hel*,*bay, which will match with anything with prefix Hel or postfix bay. 

Things that could be improved are that currently we are splitting on commas. This means in order to use commas in your message you have to insert a * instead which could break some message types. Just extra work is involved to setup in these cases.

## Option: Ignore filtered
This option when enabled will ignore messages that are being filtered. Meaning if you are filtering messages then it will not trigger a notification. On by default.

## Option: Notify trade
This option will notify you when you receive a trade request. Off by default.

## Option: Notify trade
This option will notify you when you receive a duel request. Off by default.

## Demo

waterloggedelaboratebooby-max-14mb

### CODE REVIEW

**Comment by Gamer1120** [comment]
Maybe remove the quotes from the input? "Hel*,*bay," would become Hel*,*bay. This is consistent with the ground items plugin.

**Comment by Reasel** [comment]
They are not in the actual input. They were there to show you what was in the box. Its identical to grounditems implementation.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java [hunk 1]
```diff
@@ -89,4 +90,34 @@ default int getPrayerThreshold()
 	{
 		return 0;
 	}
+
+
+	@ConfigItem(
```
Comment (SoyChai) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java]: Squash the double line break.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2018, Tanner <https://github.com/Reasel>
  * Copyright (c) 2016-2017, Abel Briggs
  * Copyright (c) 2017, Kronos <https://github.com/KronosDesign>
```
Comment (SoyChai) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: Move your name to the bottom of the attribution list, they're chronological.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 3]
```diff
@@ -408,4 +423,53 @@ private void resetTimers()
...
+	{
+		if (config.message() && isNotificationMessage(event.getMessage()))
+		{
+			notifier.notify(client.getLocalPlayer().getName() + " got message: " + event.getMessage());
+		}
+	}
+	public boolean isNotificationMessage(String message)
+	{
+		return TRUE.equals(notifiedMessages.getUnchecked(message));
+	}
```
Comment (SoyChai) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: This is correct code, but Java's native Boolean checking is pretty backwards. You can use Apache Commons' `BooleanUtils.isTrue( )` to write this more like an English sentence.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java [hunk 4]
```diff
@@ -89,4 +90,34 @@ default int getPrayerThreshold()
...
+	{
+		return "";
+	}
+
+	@ConfigItem(
+		keyName = "messageNotification",
+		name = "",
+		description = ""
+	)
+	void setMessageNotification(String key);
```
Comment (SoyChai) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java]: Will this be used by something?

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java (hunk 1 above)
Comment (Reasel) [reply runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java]: Will do.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java (hunk 4 above)
Comment (Reasel) [reply runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierConfig.java]: I was planning to add onto this to have a hotkey modifier similar to ground items for marking chat messages. But I can always add this at a later date. I will remove for now.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java (hunk 3 above)
Comment (Reasel) [reply runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: This was how it was done in GroundItems, so I wanted to stay consistent. Either way it works.

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java (hunk 2 above)
Comment (Reasel) [reply runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: Done.

**Comment by Adam-** [comment]
This seems almost identical to #2474. See my comments there.

**Comment by Reasel** [comment]
@Adam I think I have what you were looking for in the other PR. I made it into its own plugin with a few options. I can always separate the message types into two text boxes to be filled. One with Public, Clan, and Private chat and the other be things from the server like drops and such. This way is far more flexible and can accomplish the exact same thing but with less config clutter. We could even remove the trade and duel request toggle since the user could just add a "*requested trade*" message.

**Pull request closed without merging**